### ISSUE TITLE
Allow nesting based on other variables

### ISSUE THREAD

**Issue by emitanaka** [emitanaka/edibble#68]
Something like:
```r
design() %>% 
  set_units(group = 2,
                   subject = nested_in(group, 5),
                   cells = nested_in(subject, .based_on = group,
                                                           1 ~ 2,
                                                           2 ~ 4))

```

**Comment by macroscian** [comment]
Looks really useful - perhaps an even more general `nested_in(subject, ~ifelse(group=="group1",2,4)` where formulae without a lhs would be evaluated to provide the nested levels.

**Comment by emitanaka** [comment]
I agree with keeping it general like you say. Let me think about how this affects the underlying graph structure (and subsequently the suggested `design_model()`). Part of the problem is that you can end up specifying a crossed structure but it won't be encoded as such in the system and hence causes issues downstream. So in that sense, we might need some constrain so it doesn't allow users to go too wild.
